src/Sema.zig

```diff
@@ -4634,6 +4636,10 @@ fn validateUnionInit(
     var init_val: ?Value = null;
     while (block_index > 0) : (block_index -= 1) {
         const store_inst = block.instructions.items[block_index];
+        if (store_inst.toRef() == field_ptr_ref) {
+            first_block_index = block_index;
+            break;
+        }
```

Comment:
Remove the old if below this.

Reply:
Thanks, I was confident that I had deleted it.